kits/trust/contracts/TrustKit.sol

```diff
@@ -0,0 +1,342 @@
+        bytes32 financeAppId = appIds[uint8(Apps.Finance)];
+        address latestFinanceAddress = latestVersionAppBase(financeAppId);
+        return Finance(dao.newAppInstance(financeAppId, latestFinanceAddress));
+    }
+
+    function _createAgentApp(Kernel dao) internal returns (Agent) {
+        bytes32 agentAppId = appIds[uint8(Apps.Agent)];
+        address latestAgentAddress = latestVersionAppBase(agentAppId);
+        return Agent(dao.newAppInstance(agentAppId, latestAgentAddress));
+    }
+
+    function _createAgentPermissions(ACL acl, Agent agent, Voting holdVoting) internal {
+        acl.createPermission(holdVoting, agent, agent.EXECUTE_ROLE(), holdVoting);
+        acl.createPermission(holdVoting, agent, agent.RUN_SCRIPT_ROLE(), holdVoting);
+    }
+
+    function _createVotingPermissions(ACL acl, TokenManager tokenManager, Voting voting) internal {
+        acl.createPermission(tokenManager, voting, voting.CREATE_VOTES_ROLE(), voting);
+        acl.createPermission(voting, voting, voting.MODIFY_QUORUM_ROLE(), voting);
+        acl.createPermission(voting, voting, voting.MODIFY_SUPPORT_ROLE(), voting);
+    }
+
+    function _createTokenManagerPermissions(ACL acl, TokenManager tokenManager, Voting voting) internal {
+        acl.createPermission(address(this), tokenManager, tokenManager.MINT_ROLE(), address(this));
+        acl.createPermission(voting, tokenManager, tokenManager.ASSIGN_ROLE(), voting);
+        acl.createPermission(voting, tokenManager, tokenManager.REVOKE_VESTINGS_ROLE(), voting);
+    }
+
+    function _createEVMRegistryPermissions(ACL acl, Voting holdVoting) internal {
+        EVMScriptRegistry evmScriptsRegistry = EVMScriptRegistry(acl.getEVMScriptRegistry());
+        bytes32 addExecutorRole = evmScriptsRegistry.REGISTRY_ADD_EXECUTOR_ROLE();
+        acl.createPermission(holdVoting, evmScriptsRegistry, addExecutorRole, holdVoting);
+        bytes32 managerRole = evmScriptsRegistry.REGISTRY_MANAGER_ROLE();
+        acl.createPermission(holdVoting, evmScriptsRegistry, managerRole, holdVoting);
+    }
+
+    function _createFinancePermissions(ACL acl, Voting holdVoting, Finance finance, Vault vault) internal {
+        acl.createPermission(finance, vault, vault.TRANSFER_ROLE(), holdVoting);
+        acl.createPermission(holdVoting, finance, finance.CREATE_PAYMENTS_ROLE(), holdVoting);
+        acl.createPermission(holdVoting, finance, finance.EXECUTE_PAYMENTS_ROLE(), holdVoting);
+        acl.createPermission(holdVoting, finance, finance.MANAGE_PAYMENTS_ROLE(), holdVoting);
+    }
+
+    function _mintHoldTokens(TokenManager holdTokenManager, address[] beneficiaryKeys) internal {
+        holdTokenManager.mint(beneficiaryKeys[0], 1e18);
+        holdTokenManager.mint(beneficiaryKeys[1], 1e18);
+    }
+
+    function _mintHeirsTokens(TokenManager heirsTokenManager, address[] heirs, uint256[] heirsStake, uint256 blockedHeirsSupply) internal {
+        heirsTokenManager.mint(address(0), blockedHeirsSupply);
```

Comment:
As I commented before, the problem I see with this trick is that MultiSig owner should be very careful if needs to mint/burn any HEIRS token (because of lost keys, new family member or whatever) to keep the ratio 66/34.

Reply:
That's certainly a pain in the ass

Reply:
Yea totally, having early execution configurable would be awesome :)